aiogram/filters/exception.py

```diff
@@ -1,51 +1,47 @@
 import re
-from typing import Any, Dict, Pattern, Tuple, Type, Union, cast
+from typing import Any, Dict, Pattern, Type, Union, cast
 
-from pydantic import validator
-
-from aiogram.filters import BaseFilter
+from aiogram.filters.base import Filter
 from aiogram.types import TelegramObject
+from aiogram.types.error_event import ErrorEvent
 
 
-class ExceptionTypeFilter(BaseFilter):
+class ExceptionTypeFilter(Filter):
     """
     Allows to match exception by type
     """
 
-    exception: Union[Type[Exception], Tuple[Type[Exception]]]
-    """Exception type(s)"""
-
-    class Config:
-        arbitrary_types_allowed = True
+    def __init__(self, *exceptions: Type[Exception]):
+        """
+        :param exceptions: Exception type(s)
+        """
+        if not exceptions:
+            raise ValueError("At least one exception type is required")
+        self.exceptions = exceptions
 
-    async def __call__(
-        self, obj: TelegramObject, exception: Exception
-    ) -> Union[bool, Dict[str, Any]]:
-        return isinstance(exception, self.exception)
+    async def __call__(self, obj: TelegramObject) -> Union[bool, Dict[str, Any]]:
+        return isinstance(cast(ErrorEvent, obj).exception, self.exceptions)
 
 
-class ExceptionMessageFilter(BaseFilter):
+class ExceptionMessageFilter(Filter):
     """
     Allow to match exception by message
     """
 
-    pattern: Union[str, Pattern[str]]
-    """Regexp pattern"""
-
-    class Config:
-        arbitrary_types_allowed = True
-
-    @validator("pattern")
-    def _validate_match(cls, value: Union[str, Pattern[str]]) -> Union[str, Pattern[str]]:
-        if isinstance(value, str):
-            return re.compile(value)
-        return value
+    def __init__(self, pattern: Union[str, Pattern[str]]):
+        """
+        :param pattern: Regexp pattern
+        """
+        if not isinstance(pattern, str):
```

Comment:
probably `if isinstance(pattern, str)`? 
otherwise we compile here `Pattern` and not compile str